<hr/>
<h1>2. Data Analysis</h1>

<p>In this section, I will be conducting simple Data Analysis to understand the dataset provided at a deeper level. By conducting Data Analysis, I will be able to evaluate what I need to do and subsequently, path the way for other operations that need to be conducted. This section is extremely important as without doing a proper Data Analysis, I will not be able to train the Unsupervised Model effectively and may even encounter errors within the datasets due to poor planning that occured due to a poor Data Analysis conducted by me.</p>

<hr/>
<h2>2.1 Importing of CSV File</h2>

<p>Before performing Data Analysis, I will need to import the CSV file that has been provided to me. By performing the operation as shown in the cell below, I have also directly placed the CSV File into a dataframe which allows me to perform various operation on it afterwards.</p>

In [25]:
# ----------- Importing of CSV File ---------- #
df = pd.read_csv("../data/raw/CA2-Energy-Consumption-Data.csv", sep = ",")

# ----------- Execution Confirmation ----------- #
print("CSV File Successfully Imported")

CSV File Successfully Imported


<p>As shown in the cell above, the text 'CSV File Successfully Imported' shows that the previous line of code which is meant to import the CSV File has been successfully executed and that 'df' now holds the dataframe that is related to the CSV File.</p>

<hr/>
<h2>2.2 Reading of CSV File</h2>

<p>After successfully importing the CSV File as 'df', I will be taking a look at the first 10 data rows of the dataframe. I will then evaluate what needs to be done regarding the dataset that has been provided.</p>

In [29]:
# ------------- Displaying DataFrame ----------- #
df.head(10).style.background_gradient(cmap="Blues")

,DATE,Gas Consumption (tons),Electricity Consumption (MWh),Water Consumption (tons)
0,1/1/1990,18.000000,725.100000,548.800000
1,1/2/1990,15.800000,706.700000,640.700000
2,1/3/1990,17.300000,624.500000,511.100000
3,1/4/1990,18.900000,574.700000,515.300000
4,1/5/1990,22.000000,553.200000,488.400000
5,1/6/1990,23.400000,580.900000,503.200000
6,1/7/1990,24.600000,626.200000,448.300000
7,1/8/1990,24.800000,632.500000,411.700000
8,1/9/1990,23.800000,605.800000,440.800000
9,1/10/1990,20.200000,563.200000,579.000000


<p>With reference to the output of the DataFrame above, I am able to see that there is a total of 4 Data Column, 3 of which are Numerical Data and 1 of which is a Categorical Data. The Categorical Data needs to be converted into DateTime format due to the Time Series only accepting DataTime for Dates.</p>

<hr/>
<h3>2.2.1 Interpretation of CSV File's Data Column</h3>

<p><strong><u>DATE</u></strong></p>
<ul>
    <li>Represents the timestamp for each recorded entry.</li>
    <li>Indicates the period (daily, monthly, or yearly) for energy consumption measurements.</li>
</ul>

<p><strong><u>Gas Consumption (tons)</u></strong></p>
<ul>
    <li>Measures the amount of gas consumed in tons for the given date.</li>
    <li>Helps analyze trends in gas usage over time for efficiency planning.</li>
</ul>

<p><strong><u>Electricity Consumption (MWh)</u></strong></p>
<ul>
    <li>Represents the total electricity usage in megawatt-hours (MWh) for each recorded date.</li>
    <li>Used to monitor power demand and identify seasonal patterns.</li>
</ul>

<p><strong><u>Water Consumption (tons)</u></strong></p>
<ul>
    <li>Measures the water usage in tons for the given date.</li>
    <li>Useful for understanding water consumption trends and optimizing resource allocation.</li>
</ul>

<hr/>
<h2>2.3 Characteristics of Data</h2>

<p>In this sub-section, I will view the characteristics of my data such as the Numerical Characteristic for Numerical Data. I will also verify how large my dataset is and to verify the scale of the data that I am working with for Clustering.</p>

<hr/>
<h3>2.3.1 Information of Data</h3>

<p>In the cell below, I will view the informaton of the dataset. The information will include the amount of data that is not missing and also the datatype for each of the Data Column. The operation will also display the total Data Rows for the dataframe.</p>

In [37]:
# ----------- Display Information Of DataFrame ----------- #
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 397 entries, 0 to 396
Data columns (total 4 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   DATE                           397 non-null    object 
 1   Gas Consumption (tons)         397 non-null    float64
 2   Electricity Consumption (MWh)  397 non-null    float64
 3   Water Consumption (tons)       397 non-null    float64
dtypes: float64(3), object(1)
memory usage: 12.5+ KB


<p>With reference to the output of the cell above, I am able to verify that there are no Missing Data as the RangeIndex has a total of 397 entries which matches up to the Non-Null Count which also has 397. However, this will need to be confirmed through a more conventional method later on. Through the output, I am also able to see that DATE is the only Categorical Data within the entire dataset which is supported by the previous observations in the previous subsections. Therefore, the Categorical Data need to be converted to DateTime.</p>

<hr/>
<h3>2.3.2 Shape of Data</h3>

<p>From the cell below, I will execute an operation which will display the amount of Data Rows and Data Columns that my dataset contains.</p>

In [41]:
# ----------- df's Shape ----------- #
shape_data = {'Attribute': ['Rows', 'Columns'], 'Count': list(df.shape)}

# ----------- Putting Shape Into DataFrame ----------- #
df_shape = pd.DataFrame(shape_data)

# ----------- Displaying DataFrame ----------- #
df_shape.style.background_gradient(cmap="Blues")

,Attribute,Count
0,Rows,397
1,Columns,4


<p>From the cell above, I am able to verify that the operation has been successfully executed. The data is represented in the format shown below. Thus, I can verify that there is a total of 200 Data Rows and 4 Data Columns, which is supported by the previous subsection.</p>
\begin{equation}
(Row , Column)
\end{equation}

<hr/>
<h3>2.3.3 Description of Data</h3>

<p>In the cell below, I will execute an operation which will describe the Numerical Data of the dataset The information displayed will include important information, especially mean (average) which can be used as a strategy to Impute Missing Data if there are any.</p>

In [45]:
# ----------- Describing DataFrame ----------- #
df.describe().style.background_gradient(cmap="Blues")

,Gas Consumption (tons),Electricity Consumption (MWh),Water Consumption (tons)
count,397.000000,397.000000,397.000000
mean,23.785139,888.472544,484.953652
std,4.903452,153.877594,133.908863
min,11.600000,553.200000,44.400000
25%,20.200000,771.100000,384.400000
50%,23.500000,897.800000,487.400000
75%,27.900000,1005.200000,580.200000
max,46.000000,1294.000000,811.000000


<p>With reference to the output of the dataframe above, I am able to see the mean of all three Numerical Data and also see the wide range of Consumptions the dataset includes.</p>